### DISCUSSION TITLE
Age distribution

### DISCUSSION THREAD

**Discussion by Goonigh2020** [neherlab/covid19_scenarios#463] [General]
Good day,
Can you please provide status of last update of "age distribution" information in the model?
I have status for the beginning of 2020 and I can provide it if the one in the model is not the latest.
Thank you

**Comment by noleti** [comment]
Thanks for the offer! Is there a specific country where you believe the data is outdated? As far as I know, the data was downloaded from United Nations Statistics Division, and should be quite recent.

**Reply by Goonigh2020** [reply]
Sorry for not mentioning the country. 
I meant the information for Azerbaijan

**Comment by nnoll** [comment]
We pulled in all 2020 estimates from the UNSD as @noleti states so I'm not worried about them being out of date yet. It is possible I missed some country codes in the original pull (I obtained from what I thought was a complete list). If you have the data I can add it manually to our list.

**Reply by Goonigh2020** [reply]
I will be more than thankful! 
Attached you find last statistics for Azerbaijan
[Azerbaijan_pop_2020.xlsx](https://github.com/neherlab/covid19_scenarios/files/4448680/Azerbaijan_pop_2020.xlsx)

**Reply by noleti** [reply]
As far as I can tell, the numbers are fairly close to the ones we have already, with the exception of the (important) 70-79 and 80+ groups, which are 8% and 19% larger in your numbers. What is the source of your numbers?

**Reply by Goonigh2020** [reply]
I have got them from state statistical committee of Azerbaijan Republic

**Comment by rbelew** [comment]
hi,  thanks first for your fine and useful work.

i've run across your age distribution data as it was incorporated
into the Covasim individual-based simulation here:

https://github.com/InstituteforDiseaseModeling/covasim/blob/master/covasim/data/country_age_data.py

when you say "We pulled in all 2020 estimates from the UNSD," can I
confirm you are going to this source:

https://population.un.org/wpp/Download/Files/1_Indicators%20(Standard)/EXCEL_FILES/5_Interpolated/WPP2019_INT_F03_1_POPULATION_BY_AGE_ANNUAL_BOTH_SEXES.xlsx

(or the API equivalent)?  i gather this is captured via `generate_data.py --fetch` and
`sources.json`, but i can't trace it through your code?

My goal is to model  a finer-grained age stratification as it relates
to SCHOOL age children, but am exploring both individual and SEIR
models and want to stay close to your data!

**Reply by noleti** [reply]
As far as I can retrace, the "pulling" is done by running https://github.com/neherlab/covid19_scenarios/blob/master/data/scripts/download_age_dists.py, which uses a library (pandasdmx) to call the following dataset:

agency   = "UNSD"
database = "DF_UNDATA_WPP"
dataset  = "SP_POP_TOTL"

as the call takes a long time (and the result basically never changes), this is not done as part of `generate_data.py`
